src/current/v25.2/changefeed-messages.md

```diff
@@ -531,6 +531,89 @@ CREATE CHANGEFEED INTO 'scheme://sink-URI' WITH updated AS SELECT column, column
+~~~
+
+Create a changefeed that emits messages from the `customer_updates` table to Kafka and define the `route_info` column in the `headers_json_column_name` option:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+CREATE CHANGEFEED FOR TABLE customer_updates INTO 'kafka://localhost:9092' WITH headers_json_column_name = 'route_info';
+~~~
+
+The changefeed will emit each row’s `route_info` data as Kafka message headers, which Kafka brokers or stream processors can use to access the metadata without inspecting the payload.
+
+The Kafka topic receives the message payload containing the row-level change:
+
+~~~json
+{"after": {"change_description": "Updated phone number", "change_version": "v2", "customer_id": "5896dc90-a972-43e8-b69b-8b5a52691ce2", "operation_type": "update", "source_system": "crm_mobile", "update_id": "39a7bb4c-ee3b-4897-88fd-cfed94558e72", "updated_at": "2025-05-06T14:57:42.378814Z"}}
+{"after": {"change_description": "Bulk customer import", "change_version": "v1", "customer_id": "0fccf7a5-5f61-4d65-ad6b-55d8740a0874", "operation_type": "insert", "source_system": "api_batch", "update_id": "e058a098-89ba-48b0-a117-93b6b00c7eba", "updated_at": "2025-05-06T14:57:42.378814Z"}}
+{"after": {"change_description": "Changed mailing address", "change_version": "v2", "customer_id": "ccca656e-96b3-43b7-83ad-ee044627d67d", "operation_type": "update", "source_system": "support_portal", "update_id": "d0f730ce-d269-4bd5-9d2d-07c6fe65fc51", "updated_at": "2025-05-06T14:57:42.378814Z"}}
+. . .
+~~~
+
+You can query the `route_info` column to view the metadata that will be present in the Kafka headers for each corresponding message:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT 
+    update_id,
+    route_info->>'operation' AS operation,
+    route_info->>'version' AS version,
+    route_info->>'origin' AS origin
+FROM customer_updates;
+~~~
+~~~
+               update_id               | kafka_header_operation | kafka_header_version | kafka_header_origin
+---------------------------------------+------------------------+----------------------+----------------------
+  3366969e-4f9f-472d-9e07-ac65e12b1e19 | delete                 | v1                   | crm_web
+  39a7bb4c-ee3b-4897-88fd-cfed94558e72 | update                 | v2                   | crm_mobile
+  b9cd25ab-e44c-4bbc-a806-7cf701ad131e | insert                 | v1                   | crm_web
+  d0f730ce-d269-4bd5-9d2d-07c6fe65fc51 | update                 | v2                   | support_portal
+  e058a098-89ba-48b0-a117-93b6b00c7eba | insert                 | v1                   | api_batch
+(5 rows)
+~~~
+
+You may need to duplicate fields between the message envelope and headers to support efficient routing and filtering by intermediate systems, such as Kafka brokers, stream processors, or observability tools, but still maintain the full context of the change in the message for downstream applications.
```

Comment:
what is this referring to?

Reply:
can you also note use cases like distributed tracing

Reply:
Changed, and added.